spread-tests/main/user-xdg-runtime-dir-created/task.yaml

```diff
@@ -2,20 +2,39 @@ summary: Ensure that XDG_RUNTIME_DIR directory is created by snap-confine
 # This is blacklisted on debian because debian doesn't use apparmor yet
 systems: [-debian-8]
 details: |
-    This test checks that XDG_RUNTIME_DIR is actually created
+    This test checks that XDG_RUNTIME_DIR is actually created and is owned by
+    the correct user.
 prepare: |
     echo "Having installed the snapd-hacker-toolbelt snap"
     snap install snapd-hacker-toolbelt
-    echo "Having removed the XDG_RUNTIME_DIR directory"
-    rm -rf "/run/user/`id -u`/snapd-hacker-toolbelt"
+    echo "And having created a test user with a home directory"
+    useradd test
+    mkdir /home/test
+    chown test.test /home/test
 execute: |
-    cd /
-    echo "FIXME: export XDG_RUNTIME_DIR for now until snapd does it"
-    export XDG_RUNTIME_DIR="/run/user/`id -u`/snapd-hacker-toolbelt"
-    echo "We can now run snapd-hacker-toolbelt.busybox true"
-    /snap/bin/snapd-hacker-toolbelt.busybox true
+    echo "We can set a per-snap XDG_RUNTIME_DIR (until snapd does this)"
+    export XDG_RUNTIME_DIR="/run/user/$(id -u test)/snap.snapd-hacker-toolbelt"
+    echo "We can now run snapd-hacker-toolbelt.busybox true as the test user"
+    su -l -c "XDG_RUNTIME_DIR=$XDG_RUNTIME_DIR /snap/bin/snapd-hacker-toolbelt.busybox true" test
     echo "And see that the XDG_RUNTIME_DIR directory was created"
-    test -d /run/user/`id -u`/snapd-hacker-toolbelt
+    [ -d "$XDG_RUNTIME_DIR" ]
+    echo "And is owned by the right user and group"
+    [ "$(stat -c '%U.%G' /)" = root.root ]
+    [ "$(stat -c '%U.%G' /run)" = root.root ]
+    [ "$(stat -c '%U.%G' /run/user)" = root.root ]
+    [ "$(stat -c '%U.%G' /run/user/$(id -u test))" = test.test ]
+    [ "$(stat -c '%U.%G' /run/user/$(id -u test)/snap.snapd-hacker-toolbelt)" = test.test ]
+    echo "And has the right permissions"
+    [ "$(stat -c '%a' /)" = 755 ]
+    [ "$(stat -c '%a' /run)" = 755 ]
+    [ "$(stat -c '%a' /run/user)" = 755 ]
+    [ "$(stat -c '%a' /run/user/$(id -u test))" = 700 ]
+    [ "$(stat -c '%a' /run/user/$(id -u test)/snap.snapd-hacker-toolbelt)" = 755 ]
```

Comment:
This one needs to be 700. The spec says XDG_RUNTIME_DIR needs to be 700. XDG_RUNTIME_DIR for the snap is set to `/run/user/$(id -u test)/snap.name` and there might be software that verifies this.